src/createRoutine.js

```diff
@@ -1,35 +1,15 @@
-import { createAction } from 'redux-actions';
 import routineStages from './routineStages';
-
-const isFunction = value => typeof value === 'function';
+import createActionCreator from './createActionCreator';
 
 export default function createRoutine(typePrefix, payloadCreator, metaCreator) {
-  const getCreatorForType = (type, creator) => {
-    if (!creator) {
-      return creator;
-    }
-    if (isFunction(creator[type])) {
-      return creator[type];
-    }
-    if (isFunction(creator[type.toLowerCase()])) {
-      return creator[type.toLowerCase()];
-    }
-    if (isFunction(creator)) {
-      return creator;
-    }
-    return undefined;
-  };
-
-  const createActionCreator = (type) => createAction(`${typePrefix}/${type}`, getCreatorForType(type, payloadCreator), getCreatorForType(type, metaCreator));
-
   return routineStages.reduce(
     (result, stage) => {
-      const actionCreator = createActionCreator(stage);
+      const actionCreator = createActionCreator(typePrefix, stage, payloadCreator, metaCreator);
```

Comment:
Here you lost the ability to set payload/meta creators for every stage separately (e.g. one payload creator for `request` and another for `success`)

Reply:
I test payload separately creators https://github.com/afitiskin/redux-saga-routines/pull/59/commits/bae356a951a40f41b1acf283050c8e3711071070